test/with_dummyserver/test_socketlevel.py

```diff
@@ -1401,6 +1403,41 @@ def test_headers_are_sent_with_the_original_case(self):
         headers = {"foo": "bar", "bAz": "quux"}
         parsed_headers = {}
 
+        def socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b""
+            while not buf.endswith(b"\r\n\r\n"):
+                buf += sock.recv(65536)
+
+            headers_list = [header for header in buf.split(b"\r\n")[1:] if header]
+
+            for header in headers_list:
+                (key, value) = header.split(b": ")
+                parsed_headers[key.decode("ascii")] = value.decode("ascii")
+
+            sock.send(
+                ("HTTP/1.1 204 No Content\r\nContent-Length: 0\r\n\r\n").encode("utf-8")
+            )
+
+            sock.close()
+
+        self._start_server(socket_handler)
+        expected_headers = {
+            "Accept-Encoding": "identity",
+            "Host": "{0}:{1}".format(self.host, self.port),
+            "User-Agent": _DEFAULT_USER_AGENT,
+        }
+        expected_headers.update(headers)
+
+        with HTTPConnectionPool(self.host, self.port, retries=False) as pool:
+            pool.request("GET", "/", headers=HTTPHeaderDict(headers))
+            assert expected_headers == parsed_headers
+
+    def test_ua_header_can_be_overwridden(self):
+        headers = {"uSeR-AgENt": "Allstate, like Geico"}
```

Comment:
Let's remove the reference to All State and Geico.

We should also have a test case for the User-Agent being set at the `ConnectionPool` level.

Reply:
Done in the original work.